### ISSUE TITLE
[Bug-Candidate]: Echidna fails to run after specifying path to specific contract.

### ISSUE THREAD

**Issue by the-caliber** [crytic/echidna#1065]
### Describe the issue:

While running echidna with the full path to the sol file,
`echidna ./contracts/ERC721/external/test/standard/ERC721BasicTests.sol --contract TestHarness --config ./contracts/ERC721/external/test/echidna.config.yaml`  
it gives this error (described in "Relevant log output:")

But when I run directly for the whole dir without specifying the path (with `.`) it works fine.
`echidna . --contract TestHarness --config ./contracts/ERC721/external/test/echidna.config.yaml`

### Code example to reproduce the issue:

https://github.com/crytic/properties/blob/main/contracts/ERC721/internal/test/standard/ERC721BasicTests.sol

### Version:

Echidna 2.1.1
Slither 0.9.3

### Relevant log output:

```shell
└─$ echidna ./contracts/ERC721/external/test/standard/ERC721BasicTests.sol --contract TestHarness --config ./contracts/ERC721/external/test/echidna.config.yaml
[2023-06-03 16:25:15.88] Compiling ./contracts/ERC721/external/test/standard/ERC721BasicTests.sol... Done! (3.33818984s)
echidna: Couldn't compile given file
stdout:
stderr:
ERROR:CryticCompile:
Solidity version not found:
STDOUT:

STDERR:
Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/home/linuxbrew/.linuxbrew/Cellar/solc-select/1.0.3/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 84, in solc
    halt_old_architecture(path)
  File "/home/linuxbrew/.linuxbrew/Cellar/solc-select/1.0.3/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 29, in halt_old_architecture
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: solc-select is out of date. Please run `solc-select upgrade`
```

**Comment by elopez** [comment]
Hi! Echidna uses crytic-compile to perform the compilations. When you specify a folder with a hardhat/foundry/etc project (like in your "whole dir" example), the corresponding build tool is invoked to perform the compilation. In contrast, when you specify a single solidity file, the `solc` compiler is invoked directly.

In this case, it looks like your compiler is not running successfully -- it seems you need to upgrade your solc-select installation before it will run:

```
argparse.ArgumentTypeError: solc-select is out of date. Please run `solc-select upgrade`
```

Even then, getting a single `.sol` file compiled directly is usually tricky -- you will need to specify import remaps and such manually. I'd recommend you keep using the "whole directory" approach, as the compilation framework tool will handle all of that for you 👍

**Comment by the-caliber** [comment]
Thank you for the answer!
Actually, previously i tried upgrading solc-select but somehow I ended up with `urllib.error.URLError: <urlopen error [Errno 111] Connection refused>` like https://github.com/crytic/solc-select/issues/70 which is under investigation https://github.com/crytic/solc-select#connection-refused-investigation-ongoing

Currently running echidna with `.` works so i will keep using that.
Closing the issue for now as other approach works.

Thanks!
